src/Symfony/Component/HttpFoundation/ParameterBag.php

```diff
@@ -31,7 +34,11 @@ public function __construct(
     /**
      * Returns the parameters.
      *
-     * @param string|null $key The name of the parameter to return or null to get them all
+     * @template TKey of string|null
+     *
+     * @param TKey $key The name of the parameter to return or null to get them all
+     *
+     * @return (TKey is null ? array<string, mixed> : array)
```

Comment:
can be improved a bit:
```suggestion
     * @return (TKey is null ? array<string, mixed> : array<mixed>)
```